arch/x86/Kconfig

```diff
@@ -47,16 +59,97 @@ config X86_64
 	select USE_SWITCH_SUPPORTED
 	select SCHED_IPI_SUPPORTED
 	select X86_MMU
+	select X86_CPU_HAS_MMX
+	select X86_CPU_HAS_SSE
+	select X86_CPU_HAS_SSE2
+	select X86_MMX
+	select X86_SSE
+	select X86_SSE2
 
 menu "x86 Features"
 
+config X86_CPU_HAS_MMX
+	bool
+
+config X86_CPU_HAS_SSE
+	bool
+
+config X86_CPU_HAS_SSE2
+	bool
+
+config X86_CPU_HAS_SSE3
+	bool
+
+config X86_CPU_HAS_SSSE3
+	bool
+
+config X86_CPU_HAS_SSE41
+	bool
+
+config X86_CPU_HAS_SSE42
+	bool
+
+config X86_CPU_HAS_SSE4A
+	bool
+
+if FPU || X86_64
+
+config X86_MMX
+	bool "Enable MMX Support"
+	depends on X86_CPU_HAS_MMX
+	help
+	  This option enables MMX support, and the use of MMX registers
+	  by threads.
+
 config X86_SSE
 	bool "Enable SSE Support"
-	depends on FPU
+	depends on X86_CPU_HAS_SSE
 	help
 	  This option enables SSE support, and the use of SSE registers
 	  by threads.
 
+config X86_SSE2
+	bool "Enable SSE2 Support"
+	depends on X86_CPU_HAS_SSE2
+	select X86_SSE if !X86_64
```

Comment:
why this `!x86_64` condition here if you are enabling `X86_SSE`  in `config X86_64`. line 66

Reply:
Originally this was done as x86-32 requires `CONFIG_SSE` (thus `CONFIG_X86_SSE`) enabled to save SSE registers during context switch. This is not needed as x86-64 saves those registers anyway. Line 66 was later added to signal SSE/SSE2 support in x86-64, but this was never updated. So updated it.